op-chain-ops/foundry/artifact.go

```diff
@@ -162,6 +165,34 @@ type ForgeAllocs struct {
 	Accounts types.GenesisAlloc
 }
 
+// ForgeAllocs implements state.DumpAllocator, such that the EVM state can be dumped into it:
+// with a StateDB.DumpToCollector call.
+var _ state.DumpCollector = (*ForgeAllocs)(nil)
+
+func (d *ForgeAllocs) OnRoot(hash common.Hash) {
+	// Unlike the geth raw-state-dump, forge-allocs do not reference the state trie root.
+}
+
+func (d *ForgeAllocs) OnAccount(address *common.Address, account state.DumpAccount) {
+	if address == nil {
+		return
+	}
+	balance, ok := new(big.Int).SetString(account.Balance, 0)
```

Comment:
Is it possible to have an already initialized `d.Accounts[*address]` in here? It would be overridden rather than merged

Reply:
Every account should only be visited once by the dump-collector, so it shouldn't already exist. I'll add a panic to sanity-check that it doesn't, in case geth changes this, or if we misuse it.